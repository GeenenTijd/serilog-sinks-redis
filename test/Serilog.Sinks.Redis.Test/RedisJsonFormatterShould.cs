using System;
using System.Collections.Generic;
using System.IO;
using Assent;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Serilog.Events;
using Serilog.Parsing;


namespace Serilog.Sinks.Redis.Test
{
  [TestClass]
  public class RedisJsonFormatterShould
  {
    [TestMethod]
    public void MatchFormat()
    {
      var stream = new MemoryStream();
      TextWriter writer = new StreamWriter(stream);

      var config = new RedisConfiguration();
      config.MetaProperties.Add( "_index_name", "MyIndex" );
      var formatter = new RedisCompactJsonFormatter( new FormatterConfiguration(), config );
      var template = new MessageTemplateParser().Parse( "{Song}++ @{Complex}" );
      var dt = new DateTimeOffset( 2001, 2, 3, 4, 5, 6, TimeSpan.Zero );
      var properties = new List<LogEventProperty> { new LogEventProperty("Song", new ScalarValue("New Macabre")) };

      var logEvent = new LogEvent( dt, LogEventLevel.Information, null, template, properties );

      formatter.Format( logEvent, writer );

      writer.Flush();
      stream.Position = 0;
      var reader = new StreamReader( stream );
      var result = reader.ReadToEnd();
      this.Assent(result);
    }
  }
}
