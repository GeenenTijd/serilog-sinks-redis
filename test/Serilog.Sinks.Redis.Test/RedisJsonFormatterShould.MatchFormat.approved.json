{
    "Timestamp": "2001-02-03T04:05:06.0000000+00:00",
    "Level": "Information",
    "MessageTemplate": "{Song}++ @{Complex}",
    "Properties": {
        "Song": "New Macabre"
    },
    "_index_name": "MyIndex"
}
